src/multio/action/scale/Scaling.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <map>
+#include <string>
+
+#include "eckit/config/LocalConfiguration.h"
+#include "multio/config/ComponentConfiguration.h"
+#include "multio/message/Glossary.h"
+#include "multio/message/Message.h"
+
+namespace multio::action {
+
+using message::glossary;
+
+class ScaleScaling {
+private:
+    bool hasScaling_;
+    std::map<std::string, double> scaleFactor_;
+
+public:
+    explicit ScaleScaling(const config::ComponentConfiguration& compConf);
+    double getScalingFactor(const std::string paramID) const;
+    template <typename Precision>
+    void applyScaling(message::Message& msg) const {
+        if (hasScaling_) {
+            std::ostringstream key;
+            std::string cparam{"xxx"};
+
+            // Extract the scaling parameter
+            if (auto param = msg.metadata().getOpt<std::string>(glossary().param); param) {
```

Comment:
Is this a duplication of the logic from `Scale.cc`? If so then we should remove it from there. The logic here is correct, but the other comments still apply, especially about `: param`.